src/routines/stage_update.jl

```diff
@@ -1,21 +1,23 @@
 #########################TimeSeries Data Updating###########################################
-function parameter_update!(param_reference::UpdateRef{T},
-                           stage_number::Int64,
-                           sim::Simulation) where T <: PSY.Component
+function parameter_update!(
+    param_reference::UpdateRef{T},
+    container::ParameterContainer,
```

Comment:
Why do we need to pass the container here now?

Reply:
This change allows for slightly cleaner code. The call site is now iterating over containers rather than UpdateRefs, so we don't need to perform a lookup here to get the container.